File: 剑指offer/3.数组中重复的数字.py
from typing import List


class Solution:
    def findRepeatNumber(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            while nums[i] != i:
                if nums[i] == nums[nums[i]]:
                    return nums[i]
                tmp = nums[i]
                nums[i] = nums[tmp]
                nums[tmp] = tmp

File: 剑指offer/3.数组中重复的数字.cpp
#include <vector>

#include <unordered_set>

using namespace std;

class Solution {
public:
    int findRepeatNumber(vector<int>& nums)
    {
        unordered_set<int> mem;
        int ans = -1;

        for (int num : nums) {
            if (mem.count(num)) {
                ans = num;
                break;
            }
            mem.insert(num);
        }
        return ans;
    }
};